frappe/model/base_document.py

```diff
@@ -926,7 +926,9 @@ def get_msg(df, docname):
 						df.fieldname != "amended_from"
 						and (is_submittable or self.meta.is_submittable)
 						and frappe.get_meta(doctype).is_submittable
-						and DocStatus(frappe.db.get_value(doctype, docname, "docstatus") or 0).is_cancelled()
+						and DocStatus(
+							frappe.db.get_value(doctype, docname, "docstatus", cache=True) or 0
```

Comment:
This is because same submitted document can be referred many times in single document (items table)